Change sharkiqpy to sharkiq

JeffResc opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
The SharkIQ integration is currently broken, see #68624.

This PR changes the underlying pypi package from [sharkiqpy](https://pypi.org/project/sharkiqpy/) to [sharkiq](https://pypi.org/project/sharkiq/). The original maintainer of [sharkiqpy](https://pypi.org/project/sharkiqpy/) seems unreachable, so I forked the repository and made the necessary changes to fix the SharkIQ vacuum integration.

Edit: If @ajmarks continues to be unavailable, I would be interested in maintaining this, as long as no one else wants to.

### sharkiq Changelog
- Updated [`const.py`](https://github.com/JeffResc/sharkiq/blob/main/sharkiq/const.py) to reflect the new `DEVICE_URL` and `LOGIN_URL` to connect to the Shark API.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [X] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #68624
- This PR is related to issue: #68624
- Link to documentation pull request: n/a

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] The code has been formatted using Black (`black --fast homeassistant tests`)
- [X] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [X] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [X] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [X] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

JeffResc committed:
Change sharkiqpy to sharkiq

JeffResc committed:
Run gen_requirements_all

JeffResc committed:
Run hassfest

JeffResc committed:
Fix tests

TheOneOgre commented:
I agree that the pypi package should be updated due to @ajmarks unavailability since December of 2020. If possible, I also believe @JeffResc your fork should have at least 1 or 2 write-access maintainers added to it on the off-chance that you disappear like @ajmarks did.

JeffResc commented:
That sounds like a good idea to me as well. If anyone is interest in maintaining the project in the event of my disappearance, please let me know.

funkybunch commented:
@JeffResc Looks like you have https://github.com/JeffResc/sharkiq/ setup to publish to `pip` via Github Actions whenever a new release is cut?  If thats all automated I'm happy to help maintain & review PRs.

AritroSaha10 commented:
@JeffResc I'd also be willing to be a maintainer if any are still needed.

JeffResc commented:
@funkybunch @AritroSaha10 Added and added.

> Looks like you have https://github.com/JeffResc/sharkiq/ setup to publish to `pypi` via Github Actions whenever a new release is cut? If thats all automated I'm happy to help maintain & review PRs.

Yep, I used that same automation on a few other repos and it works flawlessly.

I added you both to the repo.

funkybunch reviewed (changes requested):
Since merging this PR would remove the stale `sharkiq` repo I'd recommend also updating lines `890` and `891` in `CODEOWNERS` to the maintainers of the new upstream repo: `@JeffResc @funkybunch @AritroSaha10`.

JeffResc committed:
Fix trailing newline

JeffResc committed:
Update sharkiq codeowners to new maintainers

epenet commented on homeassistant/components/sharkiq/manifest.json:
```diff
@@ -1,10 +1,10 @@
+    "codeowners": ["@ajmarks"],
+    "iot_class": "cloud_polling",
+    "loggers": ["sharkiq"]
+}
```
Since #68877 has been merged, please add a carriage return here, or it will fail

JeffResc replied:
This should be all set now

JeffResc committed:
Merge branch 'home-assistant:dev' into fix-shark

JeffResc committed:
Run prettier on files

epenet commented:
Translation files should not be adjusted manually.
They are also ignored by prettier config file so I'm not sure why it triggered an update in your commit:
https://github.com/home-assistant/core/blob/cb5c7dc716a86b7d2aa610fef386da219baa3988/.prettierignore#L4

JeffResc committed:
Revert "Run prettier on files"

This reverts commit c2d99f2501817a4bbfb29b13adec0742b31681ce.

JeffResc committed:
Run hassfest

JeffResc committed:
Fix prettier

JeffResc commented:
> Translation files should not be adjusted manually. They are also ignored by prettier config file so I'm not sure why it triggered an update in your commit:
> 
> https://github.com/home-assistant/core/blob/cb5c7dc716a86b7d2aa610fef386da219baa3988/.prettierignore#L4

Oops, I did this manually as the pre-commit hooks inside my dev container don't seem to function properly, this should be fixed

JeffResc committed:
Update sharkiq codeowners

epenet commented:
I see that the new package uses very strict dependencies:
https://github.com/JeffResc/sharkiq/blob/4136e64ff3ddfa20a576909a482ecc45bcbfd044/requirements.txt#L1-2

I am a bit worried that it might cause issues in the future, when bumping those in HomeAssistant:
```
aiohttp==3.8.1
requests==2.27.1
```

JeffResc commented:
> I see that the new package uses very strict dependencies: https://github.com/JeffResc/sharkiq/blob/4136e64ff3ddfa20a576909a482ecc45bcbfd044/requirements.txt#L1-2
> 
> I am a bit worried that it might cause issues in the future, when bumping those in HomeAssistant:
> 
> ```
> aiohttp==3.8.1
> requests==2.27.1
> ```

I normally pin the exact version that it's tested on just to ensure that nothing changes between the development environment, testing environment, and production environment, but I see what you mean here.

How do you feel about:
```
aiohttp>=3.8.1
requests>=2.27.1
```

epenet commented:
> How do you feel about:
> 
> ```
> aiohttp>=3.8.1
> requests>=2.27.1
> ```

Sounds good

epenet commented:
Separately, why is there a dependency on `requests`? Could it not be 100% async?

epenet commented:
Please rebase over #68900 to fix CI

JeffResc committed:
Merge branch 'home-assistant:dev' into fix-shark

engeles20 commented:
So what do I need to do to fix? Will the update push on it's own?

epenet reviewed (approved):
Lgtm

funkybunch commented:
> So what do I need to do to fix? Will the update push on it's own?

If your home assistant instance is pulling updates from the `dev` channel this will update automatically once this PR is merged in.  Otherwise you'll have to wait until it rolled out in a major/minor release.

If you want to update it manually see the discussion in this issue: https://github.com/home-assistant/core/issues/68624

Just know that if you make these changes manually they will be overwritten if you update home assistant 'core'.
